mlflow/store/dbmodels/models.py

```diff
@@ -0,0 +1,190 @@
+    RunStatus.FAILED,
+    RunStatus.FINISHED,
+    RunStatus.RUNNING
+]
+
+LifecycleStageTypes = [
+    RunInfo.ACTIVE_LIFECYCLE,
+    RunInfo.DELETED_LIFECYCLE
+]
+
+
+def generate_uuid():
+    return uuid.uuid4().hex
+
+
+def _validate(self):
+    if not hasattr(self, '__entity__') and self.__entity__ is not None:
+        raise Exception(
+            'sqlalchemy model <{}> needs __entity__ set'.format(self.__class__.__name__))
+
+    if not hasattr(self, '__properties__') and self.__entity__ is not None:
+        raise Exception(
+            'sqlalchemy model <{}> needs __properties__ set'.format(self.__class__.__name__))
+
+
+def _create_entity(base, model):
+
+    # create dict of kwargs properties for entity and return the intialized entity
+    config = {}
+    for k in base._properties():
+        # check if its mlflow entity and build it
+        obj = getattr(model, k)
+
+        # Run data contains list for metrics, params and tags
+        # so obj will be a list so we need to convert those items
+        if k == 'metrics':
+            obj = [Metric(o.key, o.value, o.timestamp) for o in obj]
+
+        if k == 'params':
+            obj = [Param(o.key, o.value) for o in obj]
+
+        if k == 'tags':
+            obj = [RunTag(o.key, o.value) for o in obj]
+
+        config[k] = obj
+    return base(**config)
+
+
+class SqlExperiment(Base):
+    __tablename__ = 'experiments'
+    __entity__ = Experiment
+    __properties__ = Experiment._properties()
+    experiment_id = Column(Integer, primary_key=True)
+    is_deleted = Column(Boolean, default=False)
+    name = Column(String(256), unique=True, nullable=False)
+    artifact_location = Column(Text, nullable=True)
+    lifecycle_stage = Column(Integer, default=Experiment.ACTIVE_LIFECYCLE)
+
+    __table_args__ = (
+        CheckConstraint(
+            lifecycle_stage.in_(ExperimentLifecycleStages), name='lifecycle_stage'),
+    )
+
+    def __repr__(self):
+        return '<SqlExperiment ({}, {})>'.format(self.experiment_id, self.name)
+
+    def to_mlflow_entity(self):
+        return _create_entity(Experiment, self)
+
+
+class SqlRunTag(Base):
+    __tablename__ = 'run_tag'
+    __entity__ = RunTag
+    __properties__ = RunTag._properties()
+    id = Column(Integer, primary_key=True)
+    key = Column(Text, nullable=False)
+    value = Column(Text, nullable=True)
+    run_uuid = Column(Integer, ForeignKey('run.run_uuid'))
+    run = relationship('SqlRun', backref=backref('tags', cascade='all,delete'))
+
+    def __repr__(self):
+        return '<SqlRunTag({}, {})>'.format(self.key, self.value)
+
+    def to_mlflow_entity(self):
+        return _create_entity(RunTag, self)
+
+
+class SqlMetric(Base):
+    __tablename__ = 'metric'
+    __entity__ = Metric
+    __properties__ = Metric._properties()
+    key = Column(Text, primary_key=True)
+    value = Column(Float, nullable=False)
+    timestamp = Column(Integer, default=int(time.time()), primary_key=True)
```

Comment:
instead of adding `primary_key=True` -- can we use `PrimaryKeyConstraint` to make it explicit?
Also use `BigInteger` for timestamp